proto/rill/runtime/v1/resources.proto

```diff
@@ -185,8 +186,11 @@ message MetricsViewSpec {
   message SecurityV2 {
     // Dashboard level access condition
     string access = 1;
-    // row level access condition
+    // SQL expression to apply in a WHERE clause against the underlying table
     string row_filter = 2;
+    // Dimension filter to apply to queries against the metrics view.
+    // Unlike row_filter, the query_filter filters on dimension names and is independent of the SQL dialect.
+    Expression query_filter = 5;
```

Comment:
This is being used by new mv executor and that is being used mostly in case of advanced measures, also not being used by comparison toplist query, so this will not get applied in all cases ?

Reply:
Front end will give a prompt to construct this filter to the user ?

Reply:
Yes, so the filter applies to dimensions instead of the underlying data. Dimensions can have expression and/or need unnesting, so it's more complex to template in. It's not an easy fix for the old toplist queries, will address this in a follow up by migrating everything to the new `runtime/metricsview` package.

The `runtime/metricsview` package is a new package that combines logic for constructing metrics queries and rewriting/optimizing and running them.